src/plugins/GLesmos/exportAsGLesmos.ts

```diff
@@ -18,24 +24,29 @@ export function compileGLesmos(
   concreteTree: IRExpression,
   color: string,
   fillOpacity: number,
-  id: number
-) {
+  lineOpacity: number,
+  lineWidth: number
+): GLesmosShaderPackage {
   try {
-    if (isNaN(fillOpacity)) fillOpacity = 0.4;
-    else if (fillOpacity > 1) fillOpacity = 1;
-    else if (fillOpacity < 0) fillOpacity = 0;
+    fillOpacity = preprocParam(fillOpacity, 0, 1, 0.4);
+    lineOpacity = preprocParam(lineOpacity, 0, 1, 0.9);
+    lineWidth = preprocParam(lineWidth, 0, Infinity, 2.5);
+
     const { source, deps } = emitChunkGL(concreteTree._chunk);
     const type = getGLType(concreteTree.valueType);
+
     const functionDeps: string[] = [];
     deps.forEach((d) => accDeps(functionDeps, d));
-    const f = "_f" + id.toString();
+
     return {
       deps: functionDeps.map(getDefinition),
-      defs: [`${type} ${f}(float x, float y) {\n${source}\n}`],
-      bodies: [
-        `if (${f}(x,y) > 0.0) {` +
-          `  outColor = mixColor(outColor, ${colorVec4(color, fillOpacity)});` +
-          `}`,
+      chunks: [
+        {
+          def: `${type} _f0(float x, float y) {\n    ${source}\n}`,
```

Comment:
Looks like you removed shader combining. For example, `sin(x) < [0,0.1...1]` compiles into 11 different shaders instead of just 1. This might be ok (the shader combining might have been premature optimization), but it would require testing to make sure we don't get a huge performance regression on lists of functions.

Reply:
No need to test, I already did-there is a performance hit on lists of functions now.

That said, I'm not sure if it's possible to combine everything into one shader now with the way outlines and fills need to be layered.

Will be actively thinking about this.
Maybe we could revert to old (fast) shader behavior if outlines are off?